nsscache.1
.TH NSSCACHE 1 2026-03-16 "nsscache 0.50" "User Commands"
.SH NAME
nsscache \- synchronise a local NSS cache with an upstream data source
.SH SYNOPSIS
.B nsscache
[\fIglobal options\fR] \fIcommand \fR[\fIcommand options\fR]
.SH DESCRIPTION
.B nsscache
synchronises a local NSS cache against a remote data source.
This approach allows the administrator to separate the network from
the NSS lookup codepath, improving speed and reliability of name
services.
.SH OPTIONS
Global options alter general program behaviour:
.TP
\fB\-v\fR, \fB\-\-verbose\fR
enable verbose output
.TP
\fB\-d\fR, \fB\-\-debug\fR
enable debugging output
.TP
\fB\-c\fR \fIFILE\fR, \fB\-\-config\-file\fR=\fIFILE\fR
read configuration from FILE
.TP
\fB\-\-version\fR
show program's version number and exit
.TP
\fB\-h\fR, \fB\-\-help\fR
show this help message and exit
.SH COMMANDS
.SS update
Performs an update of the configured caches from the configured sources.
.TP
\fB\-f\fR, \fB\-\-full\fR
force a full update from the data source
.TP
\fB\-\-force\fR
force the update, overriding any safeguards and checks that would
otherwise prevent the update from occurring. e.g. normally empty
results from the data source are ignored as bogus -- this option will
instruct the program to ignore its intuition and use the empty map
.TP
\fB\-m\fR \fIMAPS\fR, \fB\-\-map\fR=\fIMAPS\fR
NSS map to operate on, can be supplied multiple times
.TP
\fB\-h\fR, \fB\-\-help\fR
show help for the
.B update
command
.SS verify
Perform verification of the built caches and validation of the
system NSS configuration.
.TP
\fB\-m\fR \fIMAPS\fR, \fB\-\-map\fR=\fIMAPS\fR
NSS map to operate on, can be supplied multiple times
.TP
\fB\-h\fR, \fB\-\-help\fR
show help for the
.B verify
command
.SS status
Show the last update time of each configured cache, and other
metrics, optionally in a machine-readable format.
.TP
\fB\-m\fR \fIMAPS\fR, \fB\-\-map\fR=\fIMAPS\fR
NSS map to operate on, can be supplied multiple times
.TP
\fB\-h\fR, \fB\-\-help\fR
show help for the
.B status
command
.SS repair
Verify that the configuration is correct, that the source is
reachable, then perform a full synchronisation of the cache.
.TP
\fB\-m\fR \fIMAPS\fR, \fB\-\-map\fR=\fIMAPS\fR
NSS map to operate on, can be supplied multiple times
.TP
\fB\-h\fR, \fB\-\-help\fR
show help for the
.B repair
command
.SS help
Shows online help for each command.
.SH "SEE ALSO"
.TP
\fInsscache.conf\fP(5)
.TP
\fInsswitch.conf\fP(5)
.SH FILES
.TP
\fI\|/etc/nsscache.conf\|\fP
The system-wide configuration file
.TP
\fI\|/etc/nsswitch.conf\|\fP
The system name service switch configuration file
.SH AUTHOR
Written by Jamie Wilkinson (jaq@google.com) and Vasilios Hoffman (vasilios@google.com).
.TP
The source code lives at https://github.com/google/nsscache
.SH COPYRIGHT
Copyright \(co 2007 Google, Inc.
.br
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


debian/changelog
nsscache (0.42-1) unstable; urgency=medium

  * New upstream release.
  * Remove trailing whitespace from changelog.

 -- Jamie Wilkinson <jaq@debian.org>  Mon, 29 Mar 2021 16:05:43 +1100

nsscache (0.41-2) unstable; urgency=medium

  * Bump debian/compat to 12.
  * Upstream branch renamed to main, updated gbp.conf.
  * Do not look for the test in a relative path, fixing autopkgtest failure.
  * No slapd.pid created in the ADT test wrapper, removed.
  * Rename temp and artifact variables in autopkgtest.
  * Update debian/watch format version.
  * Bump standards version to 4.5.1.
  * Remove acute accents from manpage.

 -- Jamie Wilkinson <jaq@debian.org>  Thu, 11 Feb 2021 11:04:03 +1100

nsscache (0.41-1) unstable; urgency=medium

  * New upstream release.
  * Move slapd regression test to the top level test for project consistency.
  * Moved Debian packaging to salsa.debian.org: updated control.
  * Indicate to gbp which branch is upstream.
  * Bump standards-version to 4.4.0.
  * Update debian/compat level to 10.
  * Install conffile correctly.

 -- Jamie Wilkinson <jaq@debian.org>  Mon, 08 Feb 2021 10:52:13 +1100

nsscache (0.39-2) unstable; urgency=medium

  * Make the tests execute correctly under pybuild by overriding the default local to search.
  * Trap NoSectionError when trying to load automount prefix and suffix mutators.
  * Tell pybuild to copy in nsscache.conf as it's used as testdata for confit_test.py.

 -- Jamie Wilkinson <jaq@debian.org>  Tue, 12 Nov 2019 21:43:52 +1100

nsscache (0.39-1) unstable; urgency=medium

  * New upstream release.
  * Move the status check to the run, because beforehand the config doesn't exist.
  * Update the source options to ignore python generated files.
  * Update build deps and suggests for boto3, letting the tests pass and users who want s3 buckets to be able to succeed.
  * Update rules to run pytest tests now runtests is deleted upstream.
  * Add a config for gbp.

 -- Jamie Wilkinson <jaq@debian.org>  Tue, 12 Nov 2019 04:04:05 +1100

nsscache (0.38-2) unstable; urgency=medium

  * Ensure the autopkgtest has a bsddb dependency installed.

 -- Jamie Wilkinson <jaq@debian.org>  Sat, 09 Nov 2019 23:03:00 +1100

nsscache (0.38-1) unstable; urgency=medium

  * New upstream release.
    * Python3!  Closes: #937164

 -- Jamie Wilkinson <jaq@debian.org>  Sat, 09 Nov 2019 03:38:45 +1100

nsscache (0.37-1) unstable; urgency=medium

  * New upstream release.

 -- Jamie Wilkinson <jaq@debian.org>  Thu, 01 Nov 2018 05:29:02 +1100

nsscache (0.36-1) unstable; urgency=medium

  * New upstream release.

 -- Jamie Wilkinson <jaq@debian.org>  Sun, 22 Apr 2018 12:07:04 +1000

nsscache (0.35-1) unstable; urgency=medium

  * New upstream release.
  * Fix lintian warning; bump standards version.
  * Fix lintian warning; remove Testsuite header.

 -- Jamie Wilkinson <jaq@debian.org>  Fri, 03 Nov 2017 15:29:14 +1100

nsscache (0.34-2) unstable; urgency=medium

  * Add more diagnostics to the regtest.
  * Fix version numbers in manpages.

 -- Jamie Wilkinson <jaq@debian.org>  Tue, 18 Apr 2017 10:04:30 +1000

nsscache (0.34-1) unstable; urgency=medium

  * New upstream release.
  * Fix the path to nsscache in debci regtest, so it tests the installed
    version, not from the source directory.
  * Bump debian/compat to a supported version, take 2.

 -- Jamie Wilkinson <jaq@debian.org>  Thu, 13 Apr 2017 21:11:06 +1000

nsscache (0.33-3) unstable; urgency=high

  * Revert change to debian/compat for the RC bugfix.

 -- Jamie Wilkinson <jaq@debian.org>  Fri, 07 Apr 2017 19:19:11 +1000

nsscache (0.33-2) unstable; urgency=high (fix RC bug)

  * Update debian/links which is causing the binary to be removed.
    Closes: #857087.
  * Urgency=critical to fix RC-bug before autoremoval.
  * Update debian/compat to a supported version.

 -- Jamie Wilkinson <jaq@debian.org>  Tue, 04 Apr 2017 19:58:41 +1000

nsscache (0.33-1) unstable; urgency=medium

  * New upstream release.

 -- Jamie Wilkinson <jaq@debian.org>  Tue, 01 Nov 2016 15:32:19 +1100

nsscache (0.32-2) unstable; urgency=medium

  * Build-depend on tzdata.  Closes: #839435.
  * Bump standards version.
  * Update project homepage to GitHub.

 -- Jamie Wilkinson <jaq@debian.org>  Tue, 01 Nov 2016 14:51:20 +1100

nsscache (0.32-1) unstable; urgency=medium

  * New upstream release.

 -- Jamie Wilkinson <jaq@debian.org>  Sun, 17 Jan 2016 13:08:15 +1100

nsscache (0.30-4) unstable; urgency=medium

  * Allow stderr output in the autopkgtest test.

 -- Jamie Wilkinson <jaq@debian.org>  Sat, 30 May 2015 16:33:23 +1000

nsscache (0.30-3) unstable; urgency=medium

  * Convert to dh-python2 from python-support.

 -- Jamie Wilkinson <jaq@debian.org>  Sat, 23 May 2015 17:25:55 +1000

nsscache (0.30-2) unstable; urgency=medium

  * Add the autopkgtest testsuite back to the control file.
  * Fix some path errors in the regression test.

 -- Jamie Wilkinson <jaq@debian.org>  Thu, 21 May 2015 20:01:23 +1000

nsscache (0.30-1) unstable; urgency=medium

  * New upstream release.

 -- Jamie Wilkinson <jaq@debian.org>  Wed, 20 May 2015 08:56:19 +1000

nsscache (0.29-2) unstable; urgency=medium

  * Add dependency on libnss-db back to build-depends, to fix a FTBFS on
    amd64.  Closes: #750329.

 -- Jamie Wilkinson <jaq@debian.org>  Wed, 22 Oct 2014 10:09:50 +1100

nsscache (0.29-1) unstable; urgency=medium

  * New upstream release.
  * Update standards version.
  * Disable autopkgtest for now as the test script is broken.  Closes:
    #757019.

 -- Jamie Wilkinson <jaq@debian.org>  Tue, 21 Oct 2014 22:32:27 +1100

nsscache (0.27-3) unstable; urgency=medium

  * Updated watchfile for github.

 -- Jamie Wilkinson <jaq@debian.org>  Tue, 21 Oct 2014 19:30:07 +1100

nsscache (0.27-2) unstable; urgency=medium

  * Remove suggests: on python-zsync, which doesn't exist.

 -- Jamie Wilkinson <jaq@debian.org>  Sat, 24 May 2014 13:10:49 +1000

nsscache (0.27-1) unstable; urgency=medium

  * New upstream release.

nsscache.sh
#/usr/bin/env bash

_nsscache ()
{
    local cur prev options commands update_options other_options maps

    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"

    options='-v --verbose -d --debug -c --config-file --version -h --help'
    commands='update verify status repair help'

    update_options='-f --full --force -m --map -h --help'
    other_options='-m --map -h --help'

    maps="passwd group shadow"

    case "${COMP_CWORD}" in
        1)
            COMPREPLY=( $(compgen -W "${options} ${commands}" -- "${cur}" ))
            ;;
        2)
            case "${prev}" in
                update)
                    COMPREPLY=( $( compgen -W "${update_options}" -- "${cur}" ))
                    return 0
                    ;;
                verify|status|repair)
                    COMPREPLY=( $( compgen -W "${other_options}" -- "${cur}" ))
                    return 0
                    ;;
                -c|--config-file)
                    COMPREPLY=( $( compgen -o plusdirs -f -- "${cur}" ))
                    return 0
                    ;;
                -h|--help|--version|help)
                    return 0
                    ;;
                -v|--verbose|-d|--debug )
                    COMPREPLY=( $( compgen -W "${commands}" -- "${cur}" ))
                    return 0
                    ;;
            esac
            ;;
        3)
            case "${prev}" in
                update)
                    COMPREPLY=( $( compgen -W "${update_options}" -- "${cur}" ))
                    return 0
                    ;;
                verify|status|repair)
                    COMPREPLY=( $( compgen -W "${other_options}" -- "${cur}" ))
                    return 0
                    ;;
                -m|--map)
                    COMPREPLY=( $( compgen -W "${maps}" -- "${cur}" ))
                    return 0
                    ;;
                -f|--full|--force)
                    COMPREPLY=()
                    return 0
                    ;;
                *)
                    COMPREPLY=( $( compgen -W "${commands}" -- "${cur}" ))
                    return 0
                    ;;
            esac
            ;;
        4)
            case "${prev}" in
                update)
                    COMPREPLY=( $( compgen -W "${update_options}" -- "${cur}" ))
                    return 0
                    ;;
                verify|status|repair)
                    COMPREPLY=( $( compgen -W "${other_options}" -- "${cur}" ))
                    return 0
                    ;;
                -m|--map)
                    COMPREPLY=( $( compgen -W "${maps}" -- "${cur}" ))
                    return 0
                    ;;
                -f|--full|--force)
                    COMPREPLY=()
                    return 0
                    ;;
                *)
                    COMPREPLY=( $( compgen -W "${commands}" -- "${cur}" ))
                    return 0
                    ;;
            esac
            ;;
        5)
            case "${prev}" in
                -m|--map)
                    COMPREPLY=( $( compgen -W "${maps}" -- "${cur}" ))
                    return 0
                    ;;
            esac
            ;;
        *)
            COMPREPLY=()
            return 0
            ;;
    esac
}

complete -o filenames -F _nsscache nsscache

# ex: filetype=sh
